src/rustdoc-cache.ts

```diff
@@ -0,0 +1,92 @@
+import os = require("os");
+import hashFiles = require("hash-files");
+
+import * as exec from "@actions/exec";
+import * as path from "path";
+import * as cache from "@actions/cache";
+import * as core from "@actions/core";
+import * as rustCore from "@actions-rs/core";
+
+export class RustdocCache {
+    private readonly cachePath;
+    private readonly cargo;
+    private __cacheKey = "";
+
+    constructor(cargo: rustCore.Cargo) {
+        this.cachePath = path.join("target", "semver-checks", "cache");
```

Comment:
I wouldn't use a relative path here. Instead, I'd make the user supply a path in the constructor and then canonicalize it (make absolute + remove indirection) here before saving. Otherwise this is just a bug waiting to happen: caller constructs a cache, uses it in one place, then does a `cd` to another directory and 💥 bug.

Reply:
Definitely a good idea! I changed it to be an argument of the constructor and used `path.resolve` to convert it to the absolute path. It should be much less prone to errors now...